### PR TITLE
Support if without else

### PR DESCRIPTION
**Pull request by DanySK**
Changes:
* `if` can be now used without else if it's a statement, not an expression (i.e., not the only or the last instruction in a block)
* `let` and `=` are demoted from expressions to statements, as their use as expression is usually sign of a mistake, or of hard to read code

### CODE REVIEW

**Comment by DanySK** [comment]
This PR leads to Protelis parser 9, and in turn to a new minor of Protelis. Java 7 support will be dropped starting from such release, allowing for completing the Protelis build system maintenance.

File: protelis.parser.tests/src/org/protelis/parser/tests/ProtelisParsingTest.xtend [hunk 1]
```diff
@@ -138,4 +147,30 @@ class ProtelisParsingTest {
 		.shouldNotBeValid
 	}
 
+	@Test
+	def void testIfWithoutParentheses() {
+		'''
+		let x = if (1 < 3) { 1 };
```
Comment (jakebeal) [inline protelis.parser.tests/src/org/protelis/parser/tests/ProtelisParsingTest.xtend]: I think you also need a couple more cases that should parse:
````
let x = if (1<3) { 1 } else { 2 };
1
````
````
let foo = true;
if (foo) { 1 }
2;
````

Comment (DanySK) [reply protelis.parser.tests/src/org/protelis/parser/tests/ProtelisParsingTest.xtend]: The second is not valid due to the position of the semicolon. Converted to:
```
let foo = true;
if (foo) { 1 };
2
```

Comment (jakebeal) [reply protelis.parser.tests/src/org/protelis/parser/tests/ProtelisParsingTest.xtend]: Right, thank you for fixing that typo.

**Pull request merged**